### PR TITLE
Refactor route registration logic for improved

### PR DESCRIPTION
**Pull request by mdzahid-pro**
This commit enhances the route registration logic in the Laravel framework to support domain-specific API routes. It introduces a more streamlined approach to handle cases where a domain is provided, ensuring proper routing for API endpoints. The code has been refactored for improved readability and maintainability, offering a more robust solution for defining routes within different contexts.

### CODE REVIEW

File: src/Illuminate/Foundation/Configuration/ApplicationBuilder.php [hunk 1]
```diff
@@ -274,7 +283,7 @@ public function withCommands(array $commands = [])
     protected function withCommandRouting(array $paths)
     {
         $this->app->afterResolving(ConsoleKernel::class, function ($kernel) use ($paths) {
-            $this->app->booted(fn () => $kernel->addCommandRoutePaths($paths));
+            $kernel->setCommandRoutePaths($paths);
```
Comment (PerryvanderMeer) [inline src/Illuminate/Foundation/Configuration/ApplicationBuilder.php:286]: Please undo these changes from #50742.

Comment (mdzahid-pro) [reply src/Illuminate/Foundation/Configuration/ApplicationBuilder.php:290]: This is my first contribution in laravel if you can share more information how can which method should I undo that will help me to do this

**Comment by taylorotwell** [comment]
I am closing this pull request because it lacks sufficient explanation, tests, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!
